hibernate-core/src/main/java/org/hibernate/query/sqm/sql/internal/StandardSqmSelectTranslator.java

```diff
@@ -291,6 +294,12 @@ public ModelPart resolveModelPart(NavigablePath navigablePath) {
 					final Fetch fetch = buildFetch( fetchablePath, fetchParent, fetchable, isKeyFetchable );
 
 					if ( fetch != null ) {
+						if ( fetch.getTiming() == FetchTiming.IMMEDIATE && fetchable instanceof PluralAttributeMapping ) {
+							if ( ( (PluralAttributeMapping) fetchable ).getMappedTypeDescriptor()
+									.getCollectionSemantics() instanceof BagSemantics ) {
```

Comment:
we might well combine the above two if blocks and arguably it is more readable. Super minor for sure.